components/empty-state/empty-state-illustrated-button.js

```diff
@@ -0,0 +1,63 @@
+import '../button/button-subtle.js';
+import { bodyCompactStyles, heading2Styles } from '../typography/styles.js';
+import { emptyStateIllustratedStyles, emptyStateStyles } from './empty-state-styles.js';
+import { html, LitElement } from 'lit';
+import { getIllustration } from './getIllustration.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { runAsync } from '../../directives/run-async/run-async.js';
+
+/**
+ * The `d2l-empty-state-illustrated-button` component is an empty state component that displays an illustration and action button. The illustration property can be set to use one of the preset illustrations or a custom SVG illustration can be added in the default slot.
+ * @fires d2l-empty-state-action - Dispatched when the action button is clicked
+ * @slot - Custom SVG content if illustration property is not set
+ */
+class EmptyStateIllustratedButton extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Optional: The action text to be used in the subtle button
```

Comment:
Is the idea with making this "Optional", that the consumer could omit it, and they'd get an empty-state component without an action? It would be a little strange to use a button-specific component, only to exclude the button. I'm not sure if there is a use-case for not having an action. Worth discussing with Design.

I think this goes for the link variants as well, and confirm wether this case needs to be handled. The link variants also have properties that should seemingly be required, in the same way that `d2l-link` requires an `href`.

Reply:
That was my thought process, yes. But I agree, this is something that I had noted down to discuss with Design.